antarest/launcher/business/slurm_launcher/slurm_launcher.py

```diff
@@ -0,0 +1,241 @@
+            default_wait_time=self.config.launcher.slurm.default_wait_time,
+            default_time_limit=self.config.launcher.slurm.default_time_limit,
+            default_n_cpu=self.config.launcher.slurm.default_n_cpu,
+            studies_in_dir=str(
+                (
+                    Path(self.config.launcher.slurm.local_workspace)
+                    / "STUDIES_IN"
+                )
+            ),
+            log_dir=str(
+                (Path(self.config.launcher.slurm.local_workspace) / "LOGS")
+            ),
+            finished_dir=str(
+                (Path(self.config.launcher.slurm.local_workspace) / "OUTPUT")
+            ),
+            ssh_config_file_is_required=False,
+            ssh_configfile_path_prod_cwd=None,
+            ssh_configfile_path_prod_user=None,
+        )
+
+        parser: MainOptionParser = MainOptionParser(main_options_parameters)
+        parser.add_basic_arguments()
+        parser.add_advanced_arguments()
+        arguments = parser.parse_args([])
+
+        arguments.wait_mode = False
+        arguments.check_queue = False
+        arguments.json_ssh_config = None
+        arguments.job_id_to_kill = None
+        arguments.xpansion_mode = False
+        arguments.version = False
+        arguments.post_processing = False
+        return arguments
+
+    def _init_launcher_parameters(self) -> MainParameters:
+        main_parameters = MainParameters(
+            json_dir=self.config.launcher.slurm.local_workspace,
+            default_json_db_name=self.config.launcher.slurm.default_json_db_name,
+            slurm_script_path=self.config.launcher.slurm.slurm_script_path,
+            antares_versions_on_remote_server=self.config.launcher.slurm.antares_versions_on_remote_server,
+            default_ssh_dict_from_embedded_json={
+                "username": self.config.launcher.slurm.username,
+                "hostname": self.config.launcher.slurm.hostname,
+                "port": self.config.launcher.slurm.port,
+                "private_key_file": self.config.launcher.slurm.private_key_file,
+                "key_password": self.config.launcher.slurm.key_password,
+                "password": self.config.launcher.slurm.password,
+            },
+            db_primary_key="name",
+        )
+        return main_parameters
+
+    @staticmethod
+    def _delete_study(study_path: Path) -> None:
+        shutil.rmtree(study_path)
+
+    def _callback(
+        self, study_name: str, status: JobStatus, with_error: bool = False
+    ) -> None:
+        for callback in self.callbacks:
+            callback(study_name, status, with_error)
+
+    def _export_output(self, output_path: Path) -> BytesIO:
+        data = BytesIO()
+        zipf = ZipFile(data, "w", ZIP_DEFLATED)
+        current_dir = os.getcwd()
+        os.chdir(output_path)
+        for path in glob.glob("**", recursive=True):
+            zipf.write(path, path)
+        zipf.close()
+        os.chdir(current_dir)
```

Comment:
it would be better with a pure python version (https://docs.python.org/3/library/pathlib.html#module-pathlib) not relying on os.chdir and glob.glob